### PR TITLE
Add an early return filter for `get_upload_dir()`

### PR DESCRIPTION
**Pull request by mattwiebe**
Our use of `wp_upload_dir()` in `Activitypub\get_upload_baseurl()` was creating PHP warnings on dotcom due to its internal use of `wp_mkdir_p()` which fails there.

Fixes #

## Proposed changes:
* add a `pre_activitypub_get_upload_baseurl` filter to bypass the use of `wp_upload_dir()`

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -1478,12 +1478,22 @@ function get_attribution_domains() {
  * @return string The upload base URL.
  */
 function get_upload_baseurl() {
-	$upload_dir = \wp_get_upload_dir();
+	/**
+	 * Early filter to allow plugins to set the upload base URL.
+	 *
+	 * @param string|false $maybe_upload_dir The upload base URL or false if not set.
+	 */
+	$maybe_upload_dir = apply_filters( 'pre_activitypub_get_upload_baseurl', false );
```
Comment (pfefferle) [inline includes/functions.php:1486]: `pre_activitypub_get_upload_baseurl` or `activitypub_pre_get_upload_baseurl`???

File: includes/functions.php [hunk 2]
```diff
@@ -1478,12 +1478,22 @@ function get_attribution_domains() {
...
+	if ( false !== $maybe_upload_dir ) {
+		return $maybe_upload_dir;
+	}
+
+	$upload_dir = \wp_upload_dir();
```
Comment (pfefferle) [inline includes/functions.php]: why not `wp_get_upload_dir()`?? to save a function call?

Comment (mattwiebe) [reply includes/functions.php]: Oh, TIL. Just switching to that would have fixed our issue too.

File: includes/functions.php (hunk 1 above)
Comment (mattwiebe) [reply includes/functions.php:1486]: Core always uses a `pre_` prefix to indicate a bypass filter so I used that here.

**Comment by pfefferle** [comment]
@mattwiebe you also have to add the changelog entry to the readme.txt, then the tests are green!

File: includes/functions.php [hunk 3]
```diff
@@ -1478,12 +1478,22 @@ function get_attribution_domains() {
...
+		return $maybe_upload_dir;
+	}
+
 	$upload_dir = \wp_get_upload_dir();
 
 	/**
 	 * Filters the upload base URL.
 	 *
-	 * @param string \wp_get_upload_dir()['baseurl'] The upload base URL.
+	 * @param string \wp_upload_dir()['baseurl'] The upload base URL.
```
Comment (pfefferle) [inline includes/functions.php]:
```suggestion
	 * @param string \wp_get_upload_dir()['baseurl'] The upload base URL.
```

if you change it back, we should also change the comment

Comment (mattwiebe) [reply includes/functions.php]: Ah, nice catch

**Comment by mattwiebe** [comment]
> @mattwiebe you also have to add the changelog entry to the readme.txt, then the tests are green!

It does not like me 😆

**Comment by mattwiebe** [comment]
We could also probably just scrap the regular filter entirely. I doubt anyone was using it, since it was just for us.

**Review by pfefferle — DISMISSED** [review]

**Review by pfefferle — APPROVED** [review]

**Review by pfefferle — CHANGES_REQUESTED** [review]
~~It seems that you accidentally bring back a lot of readme stuff!~~

I quickly fixed it

**Review by pfefferle — APPROVED** [review]

**Comment by mattwiebe** [comment]
> I quickly fixed it

Thanks! The conflict resolution thing here was acting all kinds of weird for me

**Pull request merged**